### PR TITLE
Provide a descriptive quickfix/location list title.

### PR DESCRIPTION
**Pull request by datanoise**
Currently, the `w:quickfix_title` is being automatically set to the actual
Vim command that populates quickfix/location list. For example, running
any Guru command it will be displayed as `:    lgetexpr a:items` which
is not really informative.

### CODE REVIEW

**Comment by fatih** [comment]
Can you show an after/before screenshot? For example I've introduced a syntx error, but :GoFmt was not able to parse it and it errored. And a :GoBuild just displayed a normal quickfix list without any name.

**Comment by datanoise** [comment]
It depends on if you are using `%q` in your `statusline` configuration. `%q` will display the type of the quickfix window (Quickfix or Location List) and the command that was used to generate this list.

`GoBuild` displays the title correctly:

```
[Quickfix List] :go build . errors
```

but GoFmt would display:

```
[Location List] :setloclist()
```

because the location list was set explicitly with `setloclist()` function without specifying any title.

**Comment by fatih** [comment]
@datanoise This is something I want to merge definitely. I have some tasks on the vim-8.0 branch, I'll take another stab once I've merged it to master. Thanks for the work. I'll merge it if there is no problem or try to give a feedback if needed any.

**Comment by fatih** [comment]
@datanoise can you please rebase against master if possible?

File: autoload/go/guru.vim [hunk 1]
```diff
@@ -106,7 +106,7 @@ function! s:guru_cmd(args) range abort
 endfunction
 
 " sync_guru runs guru in sync mode with the given arguments
-function! s:sync_guru(args) abort
+function! s:sync_guru(args, title) abort
```
Comment (datanoise) [inline autoload/go/guru.vim]: ah, good point. I'll fix it.

File: autoload/go/guru.vim [hunk 2]
```diff
@@ -235,7 +235,7 @@ function! go#guru#Implements(selected) abort
         \ 'needs_scope': 1,
         \ }
 
-  call s:run_guru(args)
+  call s:run_guru(args, 'Implements')
```
Comment (fatih) [inline autoload/go/guru.vim]: all `run_guru` functions should be fixed as well.

**Comment by fatih** [comment]
Just tested it locally, but it doesn't work for me. This is what I got for `:GoBuild`

image

Can you please search for `Populate(...)` functions. For example there is one in `job.vim` which is the case for my error.

**Comment by datanoise** [comment]
it should be fixed now.

**Comment by fatih** [comment]
I've pulled the changes and tried `:GoMetaLinter` for the following code: https://play.golang.org/p/lZwwR-vXQl

However I don't see `:GoMetaLinter` in the title yet:

screen_shot_2016-11-24_at_10_41_04_pm

**Comment by datanoise** [comment]
I'll check it later today. I suspect that there are still cases where I didn't propagate the title.

**Comment by fatih** [comment]
Thanks @datanoise 👍

**Pull request merged**